mlrun/projects/project.py

```diff
@@ -1903,18 +2004,88 @@ def set_function(
         self._set_function(resolved_function_name, tag, function_object, func)
         return function_object
 
-    def _resolved_function(
+    def create_function(
```

Comment:
not sure why create function is needed? I think its out of context here. Why would a user want to create a function in a project and not set it?
Also, in the examples docstring you allow getting an object and a string but the type hints suggest you only accepts strings.

Reply:
Agree it somewhat out of context so removing.
Btw there is a use-case for not setting into the project - when you don't want to export it